module/zfs/zio_compress.c

```diff
@@ -178,10 +178,9 @@ zio_compress_data(enum zio_compress c, abd_t *src, void **dst, size_t s_len,
 	if (*dst == NULL)
 		*dst = zio_buf_alloc(s_len);
 
-	/* No compression algorithms can read from ABDs directly */
-	void *tmp = abd_borrow_buf_copy(src, s_len);
-	c_len = ci->ci_compress(tmp, *dst, s_len, d_len, complevel);
-	abd_return_buf(src, tmp, s_len);
+	abd_t dabd = ABD_STACK_INIT(dst, d_len);
+	c_len = ci->ci_compress(src, &dabd, s_len, d_len, complevel);
+	/* XXX abd_zero_off(&dabd, c_len, d_len - c_len); ? */
```

Comment:
I don't think anything should care about the destination ABD content after compressed data.  If something does, it should be fixed. There may likely follow a padding to the ashift or something else, but it is its duty to zero it, whether it is a ABD or plain buffer.

Reply:
Yeah, I just noticed the pattern a lot and couldn't decide. I'm persuaded, thanks!